models/vrp.rb

```diff
@@ -124,7 +126,7 @@ def self.create(hash, delete = true)
       vrp = super({})
       self.filter(hash) # TODO : add filters.rb here
       # moved filter here to make sure we do have schedule_indices (not date) to do work_day check with lapses
-      self.check_consistency(hash)
```

Comment:
It doesn't really concern the VRP, but the hash. I am not convinced at all that it should be a concern.

Reply:
What would you suggest ? We had previously created a class that validates Hash.

Reply:
We have in input a JSON received by grape, which converts it into an hash, this hash is then converted into a vrp model using active_hash.

My assumption is that, the intermediate hash should have is own class which inherit from hash and add the concern to it.
so we could directly call hash.check_consistency

Reply:
@fonsecadeline the class you proposed was completely different -- it was in Models module and inhereted from Vrp class, -- the class @braktar suggests is a lot straightforward and it is kind of a wrapper around `Hash` class so that you can add the concern to that class.

I am still not convinced that it worth the effort because the main reason we have these validations like this (i.e., manually) is that the validation functionality `validates_x_of_y` of active_hash(?) was broken in the previous Ruby version. We still haven't tested if it is still broken or not in the current Ruby version. This functionality also lets us define custom validators and it calls them automatically, and (if I am not mistaken) it gives clear fail messages with indexing etc. so our manual validators are temporary at best. 

This would make more sense if we could use this custom Hash to tell Grape to use as validator (kinda like presenter + validation) but I don't think we can do that.

Reply:
I agree that this is meant to be temporary and maybe we do not need to spend too much time trying to make it in the best way.

What should I do then ? are we okay to keep it like that ? (https://github.com/Mapotempo/optimizer-api/pull/192)